src/logic/calculate.js

```diff
@@ -0,0 +1,48 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  const { total, next, operation } = data;
+
+  // Reset all values if AC is clicked or the result/total is an error
+  if (buttonName === 'AC' || total === 'Error') {
+    return { total: null, next: null, operation: null };
+  }
+
+  // Return the percentage of the recently typed number
+  if (buttonName === '%') {
+    if (!next) return { total: (total / 100).toString, next, operation };
+    if (!total) return { total, next: (next / 100).toString, operation };
+  }
```

Comment:
You need to use the `modifier` operator from `big.js` to perform the modulus operation. 
_I strongly suggest you move this situation to the `operate.js` module_